### PR TITLE
Friedman test

### PR DESCRIPTION
**Pull request by Sarfaraz-Sayyed**
Added R code for Friedman test which is a nonparametric test for one-way repeated measures ANOVA on a simulated dataset. The code for the simulated data is also included.

### CODE REVIEW

File: R_Friedmantest.qmd [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Simulated datasets with continuous endpoints are generated for single-drug repeated measurements to check whether any significance exists between the responses at different time points. The p-value will indicate whether differences in response for different time points are significant.
+
+*Data source:*
+
+```{r, output = FALSE, message = FALSE}
+library(dplyr)
+library(rstatix) 
+```
+
+```{r eval=FALSE}
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:16]: The CAMIS repo is set up with a github action that runs the R and python code and creates the output. Therefore all the R code blocks should have eval = TRUE

```suggestion
```{r}
```

File: R_Friedmantest.qmd [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+set.seed(123)
+
+Onewayrepeat = data.frame(subject = rep(1:10, each=4),
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:19]: We often try to follow a similar style when using R. While this is a personal preference of course, the tidyverse style guide is a helpful resource. https://style.tidyverse.org/syntax.html#:~:text=Generally%2C%20variable%20names%20should%20be,this%20is%20not%20easy!).&text=Where%20possible%2C%20avoid%20re%2Dusing,the%20readers%20of%20your%20code.
Using this type of styling we would reformat your variable name as below:
```suggestion
one_way_repeat <- data.frame(subject = rep(1:10, each=4),
  timepoint = rep(c(1,2,3,4), times = 10),
  response = round(runif(n=40, 10, 50))
```

File: R_Friedmantest.qmd [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                          timepoint = rep(c(1, 2, 3, 4), times=10),
+                          Response =round(runif(n = 40, 10, 50)))
+
+head(Onewayrepeat)
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:23]:
```suggestion
head(one_way_repeat)
```

File: R_Friedmantest.qmd [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+The friedman.test() function from the R package 'rstatix' computes the following:
+
+-   Friedman chi-squared statistic
+
+-   its degrees of freedom and
+
+-   p-value for checking whether the mean response is the same for all the time points.
+
+**Missing Values**
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:36]:
```suggestion
### Missing Values
```

File: R_Friedmantest.qmd [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+when the data contain NAs. the function Defaults to getOption("na.action") which discards the row with NAs
+
+## Example Code
+
+## **Freidman Chi-Square test**
+
+```{r eval=FALSE}
+friedman.test(y=Onewayrepeat$Response, groups=Onewayrepeat$timepoint, blocks=Onewayrepeat$subject)
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:45]: update variables names as above including changing the `Response` variable to `response`

File: R_Friedmantest.qmd [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+where: y - a vector of response values. groups - a vector of values indicating the "group" an observation belongs in. blocks - a vector of values indicating the "blocking" variable.
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:48]: Perhaps earlier when you describe the simulated data generation you could relate it back to these params?

File: R_Friedmantest.qmd [hunk 7]
```diff
@@ -0,0 +1,52 @@
+---
+title: "Friedman Chi-Square test using R package rstatix"
+---
+
+# **Example: Simulated dataset with 10 subjects each having results at each of the 4 time points.**
+
+Simulated datasets with continuous endpoints are generated for single-drug repeated measurements to check whether any significance exists between the responses at different time points. The p-value will indicate whether differences in response for different time points are significant.
+
+*Data source:*
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:9]: Remove?

File: R_Friedmantest.qmd [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+**Missing Values**
+
+when the data contain NAs. the function Defaults to getOption("na.action") which discards the row with NAs
+
+## Example Code
+
+## **Freidman Chi-Square test**
+
+```{r eval=FALSE}
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:44]:
```suggestion
```{r}
```

File: R_Friedmantest.qmd [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+```{r eval=TRUE}
+library(dplyr)
+library(rstatix) 
+```
+
+```{r eval=TRUE}
+set.seed(123)
+
+One_way_repeat <- data.frame(subject = rep(1:10, each=4),
```
Comment (orladoylenvs) [inline R_Friedmantest.qmd:24]: I wouldn't recommend that you use a capital letter for your variable name. Of course this doesn't affect the quality and is a bit pedantic :)

Comment (Sarfaraz-Sayyed) [reply R_Friedmantest.qmd]: Thanks for all the review and comments. Have updated and pushed again.

**Comment by orladoylenvs** [comment]
Thanks @Sarfaraz-Sayyed for making the updates!! 

@statasaurus ok to have the SAS and R implementations in the same PR?

**Pull request merged**